gravitee-am-ui/src/app/domain/applications/application/advanced/factors/mfa/factors-select-dialog/factors-select-dialog.component.ts

```diff
@@ -70,8 +70,9 @@ export class FactorsSelectDialogComponent implements OnInit {
 
   goToMfaFactorSettingsPage(event: any) {
     event.preventDefault();
-    const url = `/environments/${this.data.environment}/domains/${this.data.domainName}/settings/factors`;
-    window.open(url, '_blank');
+    const domainUrlIndex = this.data.currentUrl.indexOf(this.data.domainUrl);
+    const basePath = this.data.currentUrl.slice(0, domainUrlIndex);
```

Comment:
it would suggest moving the logic of building the path to the parent component, getting it in ngOnInit in this file and here it could just passed to open(...) function